src/core.js

```diff
@@ -0,0 +1,661 @@
+     * @property {Array<Array>} open - An array of the arguments passed when called the mock's `opeen` method.
+     * @property {Array<Array>} close - An array of the arguments passed when called the mock's `close` method.
+     */
+
+    /**
+     * @typedef MIDIPort
+     * @type {Object}
+     * @property {String} id - The unique ID of the device associated with the port.
+     * @property {String} name - The name of the device associated with the port.
+     * @property {String} manufacturer - The manufacturer of the device associated with the port.
+     * @property {String} type - The type of port, either "input", or "output".
+     * @property {String} state - The state of the port, either "open", "pending", or "closed".
+     * @property {String} connected - "connected" if the port is connected, "disconnected" otherwise.
+     * @property {Function} send - A function that handles requests to send messages to this port.
+     * @property {Function} clear - A function that handles requests to clear messages in progress being sent by this port.
+     * @property {Function} open - A function that will handle requests to open a connection to this port.
+     * @property {Function} close - A function that will handle requests to close the connection to this port.
+     */
+
+    /** @typedef MIDIInput
+     * @type {MIDIPort}
+     * @property {AddEventListener} addEventListener - Add a listener for events the input receives.
+     * @property {DispatchEvent} dispatchEvent - Dispatch an event to this input.
+     */
+
+    /**
+     * @typedef MIDIOutput
+     * @type {MIDIPort}
+     * @property {Function} send - A function that handles requests to send messages to this port.
+     * @property {Function} clear - A function that handles requests to clear messages in progress being sent by this port.
+     */
+
+    youme.requestAccess = function (sysex, software, onAccessGranted, onError) {
+        if (!navigator.requestMIDIAccess) {
+            var msg = "The Web MIDI API is not available. You may need to enable it in your browser's settings.";
+            fluid.log(fluid.logLevel.WARN, msg);
+            onError(msg);
+            return;
+        }
+
+        var p = navigator.requestMIDIAccess({
+            sysex: sysex,
+            software: software
+        });
+
+        p.then(onAccessGranted, onError);
+    };
+
+    youme.getPorts = function (access) {
+        var ports = {};
+
+        youme.collectPorts("inputs", access, ports);
+        youme.collectPorts("outputs", access, ports);
+
+        return ports;
+    };
+
+    youme.requestPorts = function (success, error, software) {
+        function wrappedSuccess(access) {
+            var ports = youme.getPorts(access);
+            success(ports);
+        }
+
+        youme.requestAccess(false, software, wrappedSuccess, error);
+    };
+
+    youme.createPortViews = function (portsArray) {
```

Comment:
In the new world, perhaps "portsToViews" or some similar kind of name?

Reply:
I used your suggested name.  I think this is only used in the port logging stuff, but still.